app/models/export/activity_comments_column.rb

```diff
@@ -0,0 +1,42 @@
+class Export::ActivityCommentsColumn
+  def initialize(activities:, report: nil)
+    @activities = activities
+    @report = report
+  end
+
+  def headers
+    ["Comments in report"]
+  end
+
+  def rows
+    return [] if @activities.empty?
+
+    @activities.map { |activity|
+      [activity.id, comments_for_activity(activity).join("\n----\n")]
+    }.to_h
+  end
+
+  private
+
+  def all_comments_for_report
+    @_comments ||= @report.comments.includes(:commentable)
+  end
+
+  def all_comments
+    @_comments ||= Comment.all.includes(:commentable)
```

Comment:
Are we able to constrain this `all_comments` branch in some way? If the exporter is BEIS, then is the export for one organisation at a time? Or is there the ability to download every single comment in one giant export? 

Is it maybe worth profiling memory usage based on some much larger quantity of comments (an estimate of what might be in RODA in a couple of years)?

Maybe there's no problem and this is all using lazy loading cleverness? I haven't had a chance to play with it...

Reply:
To be honest, we don't have a use for the all right now - I can take it out and still deliver the report csv - I'll do that for now.

Reply:
All of the export parts are very expensive - I would love to have some time to look at how they are performing in terms of memory (and other metrics) - not sure we will get the opportunity to do so though.

I like the idea that this class is not concerned with constraints - perhaps that is naive of me, but I like the idea that the responsibility of this is simply to take the activities and give you back the comments for export as tabular data, as a developer it is our responsibility to understand the impact of doing so for all activities and the cost of doing so.

Another one for a good discussion over coffee? :)